### ISSUE TITLE
libopenblasp C extension error

### ISSUE THREAD

**Issue by adamgreenhall** [conda/conda#2405]
Getting a new error after months of using these environment specs. Any ideas on how to fix (aside from upgrading packages)?

```
ImportError: C extension: libopenblasp-r0-ea933816.2.18.so: cannot open shared object file: No such file or directory not built. If you want to import pandas from the source directory, you may need to run 'python setup.py build_ext --inplace' to build the C extensions first.
```

 Here are the full details:

```
$ service_venv conda list
# packages in environment at /srv/conda:
#
bottleneck                1.0.0               np110py27_0
cairo                     1.12.18                       6
conda                     4.0.5                    py27_0
conda-env                 2.4.5                    py27_0
configparser              3.5.0b2                  py27_1
cycler                    0.10.0                   py27_0
cython                    0.24                     py27_0
decorator                 4.0.9                    py27_0
entrypoints               0.2                      py27_1
flask                     0.10.1                   py27_1
fontconfig                2.11.1                        5
freetype                  2.5.5                         0
funcsigs                  1.0.0                    py27_0
futures                   3.0.5                    py27_0
geos                      3.3.3                         0
get_terminal_size         1.0.0                    py27_0
gevent                    1.1.1                    py27_0
greenlet                  0.4.9                    py27_0
ipykernel                 4.3.1                    py27_0
ipython                   4.2.0                    py27_0
ipython-notebook          4.0.4                    py27_0
ipython_genutils          0.1.0                    py27_0
ipywidgets                4.1.1                    py27_0
itsdangerous              0.24                     py27_0
jinja2                    2.8                      py27_0
jsonschema                2.4.0                    py27_0
jupyter                   1.0.0                    py27_2
jupyter_client            4.2.2                    py27_0
jupyter_console           4.1.1                    py27_0
jupyter_core              4.1.0                    py27_0
libgfortran               3.0                           0
libpng                    1.6.17                        0
libsodium                 1.0.3                         0
libxml2                   2.9.2                         0
line-profiler             1.0                       <pip>
lmdb                      0.84                      <pip>
lockfile                  0.12.2                    <pip>
luigi                     2.1.1                     <pip>
lyft-marketplace-stdlib   0.18.0                    <pip>
lyft-stdlib               1.23.2                    <pip>
markupsafe                0.23                     py27_0
matplotlib                1.5.1               np110py27_0
mistune                   0.7.2                    py27_0
mkl                       11.3.1                        0
pyzmq                     14.7.0                   py27_1
qt                        4.8.7                         1
qtconsole                 4.2.1                    py27_0
readline                  6.2                           2
redis                     2.10.5                    <pip>
redshift-sqlalchemy       0.4.1                     <pip>
requests                  2.9.1                    py27_0
runipy                    0.1.3                    py27_0
scikit-learn              0.17.1              np110py27_0
scipy                     0.17.0              np110py27_2
seaborn                   0.7.0                    py27_0
setuptools                20.7.0                   py27_0
shapely                   1.5.1                    py27_0
simplegeneric             0.8.1                    py27_0
singledispatch            3.4.0.3                  py27_0
sip                       4.16.9                   py27_0
six                       1.10.0                   py27_0
smmap                     0.9.0                     <pip>
sqlalchemy                1.0.12                   py27_0
sqlite                    3.9.2                         0
ssl_match_hostname        3.4.0.2                  py27_1
statsd                    3.0                       <pip>
statsmodels               0.6.1               np110py27_0
tables                    3.2.2                     <pip>
terminado                 0.5                      py27_1
tk                        8.5.18                        0
tornado                   4.3                      py27_0
traitlets                 4.2.1                    py27_0
ujson                     1.35                     py27_0
werkzeug                  0.11.8                   py27_0
wheel                     0.29.0                   py27_0
yaml                      0.1.6                         0
zeromq                    4.1.3                         0
zlib                      1.2.8                         0

$ service_venv python
Python 2.7.11 |Continuum Analytics, Inc.| (default, Dec  6 2015, 18:08:32)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import pandas as pd
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/conda/lib/python2.7/site-packages/pandas/__init__.py", line 13, in <module>
    "extensions first.".format(module))
ImportError: C extension: libopenblasp-r0-ea933816.2.18.so: cannot open shared object file: No such file or directory not built. If you want to import pandas from the source directory, you may need to run 'python setup.py build_ext --inplace' to build the C extensions first.
```

**Comment by adamgreenhall** [comment]
I suspect this has something to do with getting force upgraded from 3.19 -> 4.0:

    $ conda install ...
    The following packages will be UPDATED:

        conda: 3.19.0-py27_0 --> 4.0.5-py27_0

**Comment by adamgreenhall** [comment]
Failure to support older packages and unexpected, hard to debug changes like this are making us consider moving completely off of conda ASAP. Any ideas what's going on here?

**Comment by msarahan** [comment]
Pandas is not present in your ```conda info``` output above.  That means you installed pandas with pip, or possibly from source.  We can't possibly anticipate those use cases.  We do our best to ship working packages, though.  Did you try to ```conda install pandas```?  If not, why not?

**Comment by adamgreenhall** [comment]
sorry - I think that ``conda info`` was from an in-between debugging state. here's the original (from a travis CI run). in this one pandas is conda installed. the other thing I notice is that conda is upgrading itself to 4.0 during install (we had issues like this with 4.0 when it first came out and have had conda pinned to 3.18 since).  

    Worker information
    hostname: travis-worker-gce-com-prod-3:fac1f3c9-f75c-4685-9f8f-e7f0643f03d5
    OS name: "linux", version: "3.13.0-63-generic", arch: "amd64", family: "unix"


    $ python --version
    Python 2.7.9
    $ pip --version
    pip 6.0.7 from /home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages (python 2.7)
    $ wget http://repo.continuum.io/miniconda/Miniconda2-3.18.3-Linux-x86_64.sh -O miniconda.sh
    --2016-04-27 16:32:00--  http://repo.continuum.io/miniconda/Miniconda2-3.18.3-Linux-x86_64.sh
    Resolving repo.continuum.io (repo.continuum.io)... 23.23.167.24, 184.73.249.242, 184.72.230.63, ...
    Connecting to repo.continuum.io (repo.continuum.io)|23.23.167.24|:80... connected.
    HTTP request sent, awaiting response... 200 OK
    Length: 23460669 (22M) [application/octet-stream]
    Saving to: `miniconda.sh'

    100%[======================================>] 23,460,669  18.9M/s   in 1.2s

    2016-04-27 16:32:01 (18.9 MB/s) - `miniconda.sh' saved [23460669/23460669]


    $ chmod +x miniconda.sh

    $ ./miniconda.sh -b
    PREFIX=/home/travis/miniconda2
    installing: python-2.7.10-2 ...
    installing: conda-env-2.4.4-py27_0 ...
    installing: openssl-1.0.2d-0 ...
    installing: pycosat-0.6.1-py27_0 ...
    installing: pyyaml-3.11-py27_1 ...
    installing: readline-6.2-2 ...
    installing: requests-2.8.1-py27_0 ...
    installing: sqlite-3.8.4.1-1 ...
    installing: tk-8.5.18-0 ...
    installing: yaml-0.1.6-0 ...
    installing: zlib-1.2.8-0 ...
    installing: conda-3.18.3-py27_0 ...
    installing: pycrypto-2.6.1-py27_0 ...
    installing: _cache-0.0-py27_x0 ...
    Python 2.7.10 :: Continuum Analytics, Inc.
    creating default environment...
    installation finished.

    $ export PATH=/home/travis/miniconda2/bin:$PATH

    $ conda install --yes python=$TRAVIS_PYTHON_VERSION --file conda-requirements.txt
    Fetching package metadata: ....
    Solving package specifications: ................................
    Package plan for installation in environment /home/travis/miniconda2:

    The following packages will be downloaded:

        package                    |            build
        ---------------------------|-----------------
        mkl-11.3.1                 |                0       121.2 MB
        openssl-1.0.2g             |                0         3.2 MB
        sqlite-3.9.2               |                0         3.9 MB
        python-2.7.11              |                0        12.0 MB
        conda-env-2.4.5            |           py27_0          24 KB
        nose-1.3.7                 |           py27_0         194 KB
        numpy-1.9.3                |           py27_2         5.7 MB
        pytz-2016.3                |           py27_0         178 KB
        requests-2.9.1             |           py27_0         605 KB
        setuptools-20.7.0          |           py27_0         453 KB
        six-1.10.0                 |           py27_0          16 KB
        wheel-0.29.0               |           py27_0          81 KB
        conda-4.0.5                |           py27_0         185 KB
        pip-8.1.1                  |           py27_1         1.5 MB
        python-dateutil-2.5.2      |           py27_0         236 KB
        scipy-0.15.1               |       np19py27_0        37.0 MB
        pandas-0.16.2              |       np19py27_0        10.8 MB
        ------------------------------------------------------------
                                               Total:       197.1 MB

    The following NEW packages will be INSTALLED:

        mkl:             11.3.1-0
        nose:            1.3.7-py27_0
        numpy:           1.9.3-py27_2
        pandas:          0.16.2-np19py27_0
        pip:             8.1.1-py27_1
        python-dateutil: 2.5.2-py27_0
        pytz:            2016.3-py27_0
        scipy:           0.15.1-np19py27_0
        setuptools:      20.7.0-py27_0
        six:             1.10.0-py27_0
        wheel:           0.29.0-py27_0

    The following packages will be UPDATED:

        conda:           3.18.3-py27_0 --> 4.0.5-py27_0           # surprise!!!! force upgrade
        conda-env:       2.4.4-py27_0  --> 2.4.5-py27_0
        openssl:         1.0.2d-0      --> 1.0.2g-0
        python:          2.7.10-2      --> 2.7.11-0
        requests:        2.8.1-py27_0  --> 2.9.1-py27_0
        sqlite:          3.8.4.1-1     --> 3.9.2-0

    Fetching packages ...
    mkl-11.3.1-0.t 100% |################################| Time: 0:00:02  51.63 MB/s
    openssl-1.0.2g 100% |################################| Time: 0:00:00   4.21 MB/s
    sqlite-3.9.2-0 100% |################################| Time: 0:00:00   5.14 MB/s
    python-2.7.11- 100% |################################| Time: 0:00:01  10.78 MB/s
    conda-env-2.4. 100% |################################| Time: 0:00:00 880.67 kB/s
    nose-1.3.7-py2 100% |################################| Time: 0:00:00 994.41 kB/s
    numpy-1.9.3-py 100% |################################| Time: 0:00:00   6.17 MB/s
    pytz-2016.3-py 100% |################################| Time: 0:00:00   1.04 MB/s
    requests-2.9.1 100% |################################| Time: 0:00:00   1.80 MB/s
    setuptools-20. 100% |################################| Time: 0:00:00   1.49 MB/s
    six-1.10.0-py2 100% |################################| Time: 0:00:00 588.18 kB/s
    wheel-0.29.0-p 100% |################################| Time: 0:00:00 735.66 kB/s
    conda-4.0.5-py 100% |################################| Time: 0:00:00   1.12 MB/s
    pip-8.1.1-py27 100% |################################| Time: 0:00:00   3.27 MB/s
    python-dateuti 100% |################################| Time: 0:00:00   1.22 MB/s
    scipy-0.15.1-n 100% |################################| Time: 0:00:01  26.15 MB/s
    pandas-0.16.2- 100% |################################| Time: 0:00:00  11.43 MB/s
    Extracting packages ...
    [      COMPLETE      ]|###################################################| 100%
    Unlinking packages ...
    [      COMPLETE      ]|###################################################| 100%
    Linking packages ...
    [      COMPLETE      ]|###################################################| 100%

    $ /home/travis/miniconda2/bin/pip install -e .[tests]
    [install private package here]

    $ conda list
    # packages in environment at /home/travis/miniconda2:
    #
    conda                     4.0.5                    py27_0
    conda-env                 2.4.5                    py27_0
    flake8                    2.2.3                     <pip>
    haversine                 0.1                       <pip>
    mkl                       11.3.1                        0
    nose                      1.3.7                    py27_0
    numpy                     1.9.3                    py27_2
    openssl                   1.0.2g                        0
    pandas                    0.16.2               np19py27_0
    pep8                      1.7.0                     <pip>
    pip                       8.1.1                    py27_1
    psutil                    2.1.3                     <pip>
    pycosat                   0.6.1                    py27_0
    pycrypto                  2.6.1                    py27_0
    pyflakes                  0.8.1                     <pip>
    python                    2.7.11                        0
    python-dateutil           2.5.2                    py27_0
    pytz                      2016.3                   py27_0
    pyyaml                    3.11                     py27_1
    readline                  6.2                           2
    requests                  2.9.1                    py27_0
    scipy                     0.15.1               np19py27_0
    setuptools                20.7.0                   py27_0
    six                       1.10.0                   py27_0
    sqlite                    3.9.2                         0
    tk                        8.5.18                        0
    wheel                     0.29.0                   py27_0
    yaml                      0.1.6                         0
    zlib                      1.2.8                         0

    $ /home/travis/miniconda2/bin/nosetests
    E..
    ======================================================================
    ERROR: Failure: ImportError (libgfortran.so.1: cannot open shared object file: No such file or directory)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/travis/miniconda2/lib/python2.7/site-packages/nose/loader.py", line 414, in loadTestsFromName
        addr.filename, addr.module)
      File "/home/travis/miniconda2/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
        return self.importFromDir(dir_path, fqname)
      File "/home/travis/miniconda2/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
        mod = load_module(part_fqname, fh, filename, desc)
        ...
        from scipy import signal
      File "/home/travis/miniconda2/lib/python2.7/site-packages/scipy/signal/__init__.py", line 264, in <module>
        from .bsplines import *
      File "/home/travis/miniconda2/lib/python2.7/site-packages/scipy/signal/bsplines.py", line 4, in <module>
        import scipy.special
      File "/home/travis/miniconda2/lib/python2.7/site-packages/scipy/special/__init__.py", line 586, in <module>
        from ._ufuncs import *
    ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory

    ----------------------------------------------------------------------
    Ran 3 tests in 0.010s

    FAILED (errors=1)

**Comment by msarahan** [comment]
libgfortran errors are because we had to update our compiler to fix some mathematical errors that were showing up in scipy.  Many packages in our system do not appropriately pin libgfortran to the older version, regrettably leading to the current situation: the newer libgfortran breaks old packages.

You can try either:
- updating your software to use Numpy 1.10 or above (in your conda-requirements.txt).  All numpy 1.10 builds should use the newer library, which should be a dependency where necessary (not sure actually why it isn't a dependency for you already)
- adding libgfortran ==1.* to your conda requirements

Regarding the automatic updating, there has been a lot of discussion about changing this, but no traction yet.  One very important difference between conda 3.x and 4.x is that 4.x no longer allows inconsistent package dependencies.  This is both good and bad.  It means that when things are built properly and dependencies line up, things work more reliably.  It also means that environments that used to work (sometimes suboptimally) may no longer be installable, with unsatisfiable dependency issues.

**Comment by adamgreenhall** [comment]
thanks for the context. I think the root cause is that we're still on ``pandas==0.16``, which is not compatible (in conda at least) with ``numpy==1.10``. Will try out moving to: ``conda==4.x, pandas==0.17, numpy==1.10``

**Comment by groutr** [comment]
@adamgreenhall many of the issues with conda 4.0 have been resolved by conda 4.0.5.
